cmake/DetectOptions.cmake

```diff
@@ -177,22 +177,30 @@ endif()
 # Kokkos
 if(ADIOS2_USE_Kokkos)
   if(ADIOS2_USE_Kokkos STREQUAL AUTO)
-    find_package(Kokkos 3.7...<4.0 QUIET)
+    find_package(Kokkos QUIET)
   else()
-    find_package(Kokkos 3.7...<4.0 REQUIRED)
+    find_package(Kokkos REQUIRED)
   endif()
   if(Kokkos_FOUND)
-    set(ADIOS2_HAVE_Kokkos TRUE)
-    if(Kokkos_ENABLE_CUDA OR Kokkos_ENABLE_HIP OR Kokkos_ENABLE_SYCL)
-      if(Kokkos_ENABLE_CUDA)
-        set(ADIOS2_HAVE_Kokkos_CUDA TRUE)
-        enable_language(CUDA)
+    # There is no easy way in CMake find_package with a range of multiple minor versions.
```

Comment:
Why would `find_package(Kokkos 3.7 REQUIRED)` not work? Why do we need to separate this.

Reply:
3.7 will work with only 3.7, we want to support kokkos >=3.7.

On the other hand the range 3.7...4.0 does not really work as we expected it only works  between patch releases 3.7.0...3.7.4; for diff minor releases the ...< is possible but still for only symbolizing until next minor release. We want to support 3.7 and 4.0 and possibly newer versions

Reply:
I am building ADIOS with Kokkos 4.0 with `find_package(Kokkos 3.7 REQUIRED)` and it works fine. In our case the 3.7 is the minimum requirement because Kokkos is forward compatible:
https://github.com/kokkos/kokkos/blob/62d2b6c879b74b6ae7bd06eb3e5e80139c4708e6/cmake/kokkos_install.cmake#L20-L22